app/controllers/api/v2/app_groups_controller.rb

```diff
@@ -205,11 +205,71 @@ def deactivated_by_cluster_name
     }, status: :ok and return
   end
 
+  def fetch_redact_labels
+    valid, error_response = validate_required_keys(
+      [:app_group_secret])
+    render json: error_response, status: error_response[:code] and return unless valid
+
+    all_labels = {}
```

Comment:
how often the consumer will hit this Endpoint API? we can store the result in cache for 5/10 minutes and check the cache first before check directly to DB. e.g to check from cache first can follow this [link](https://github.com/BaritoLog/BaritoMarket/blob/master/app/controllers/api/v2/apps_controller.rb#L8-L15)

We can also implement the same think on consumer side

Reply:
added caching on [this](https://github.com/BaritoLog/BaritoMarket/pull/298/commits/2bf22c26ae2dacc5e5a6609cf72aa740a2afde3d) commit